src/ngx_http_lua_module.c

```diff
@@ -545,6 +553,16 @@ ngx_http_lua_init(ngx_conf_t *cf)
         }
     }
 
+    /*add the cleanup of semaphores after the lua_close */
+    cln = ngx_pool_cleanup_add(cf->pool, 0);
+    if (cln == NULL) {
+        return NGX_ERROR;
+    }
+    cln->data = NULL;
```

Comment:
It's a shame that we do not utilize this `data` pointer but rely on global C variables.